### DISCUSSION TITLE
Quest priority list and removing quest dependecies

### DISCUSSION THREAD

**Discussion by lynxdingo** [streetcomplete/StreetComplete#3788] [General]
I have noticed when I deactivate certain quests all of their child quests disappear with them. Like I would like to map buildings in this order:

- building house number
- building height
- building roof shape
- building type

Though if I deactivate the "building type" quest, I cannot do the other quests either. I cannot tell if this is a feature or bug.

Additionally, I noticed that, although one can move the order of the quests in the list, it doesn't do anything in how they show up on the map. Hiding them would be an option, but then it is just too many clicks to bring back the quests.

It has been discussed in the [other discussion](https://github.com/streetcomplete/StreetComplete/discussions/3558) on the pesky "building type" quests, thus putting it at the rear of the priority list would be a time saver and personally also less invasive when figuring the value out. From afar one can easily see a building height/roof, though more times than not, one has to walk up to the entrance of the building to figure out if one is dealing with an apartment or house.

**Comment by matkoniecz** [comment]
> Though if I deactivate the "building type" quest, I cannot do the other quests either. I cannot tell if this is a feature or bug.

To be more exact: you can solve this quest, but only for objects that are already eligible.

It is intended, otherwise app would ask for address of every single `building=yes`, including sheds and garages.

> although one can move the order of the quests in the list, it doesn't do anything in how they show up on the map

If building is eligible for address quest (mostly: it has no address set, is in country where this quest is not blocked and has eligible `building` value such as `building=house`) then it will be displayed with higher priority than later quests. By default, with priority higher than building type quests.

But if one of that is not fulfilled then it will be not eligible. Specifically `building=yes` objects will display building type quest and after answering with something that makes it eligible for other quests it will be shown.

https://wiki.openstreetmap.org/wiki/ID or https://wiki.openstreetmap.org/wiki/JOSM may be useful if your area has hundreds of obvious houses/apartment buildings - all marked as `building=yes`

**Reply by lynxdingo** [reply]
Thanks for the clarification, it makes sense.

Although JOSM is ideal for mapping a lot of amenities quickly, it doesn't help in the houses/apartment department, since from afar it looks like a house, though on closer inspection it actually is an apartment

**Reply by mnalis** [reply]
marking them as more generic `building=residential` is one workaround for that specific issue

**Reply by lynxdingo** [reply]
The nice thing of that workaround is that technically there in no vandalism, though in hindsight one should probably revert the tag back to `building=yes`.

**Comment by mnalis** [comment]
- See https://github.com/streetcomplete/StreetComplete/issues/2464 (currently paused until Layers are implemented) for more in-depth discussion on the problem and possible solutions
